docs/pages/bare/using-expo-client.md

```diff
@@ -53,121 +73,67 @@ if (Constants.appOwnership === 'expo') {
-// provide TypeScript typing and validation.
+1. You need to know the native module name ahead of time.
+2. You likely want an error to be thrown when a native module is missing in a custom build, this helps you determine if there was a native linking issue.
 
-export function logEvent(options) {
-  MyAnalytics.logEvent(options);
-}
+## Optional imports
 
-export function setUser(options) {
-  MyAnalytics.setUser(options);
-}
-```
-
-### Alternatively, use optional imports
-
-An alternative approach to using `expo-constants` as described above in the [conditional inline requires section](#use-conditional-inline-requires-to-provide-fallbacks) is to use `try/catch` around `require`. There's not any particularly good reason for you to use this in your application code, and it may lead to warning messages when the library throws an error on importing, but it is listed here in case you find yourself with a fitting use case.
+[Metro bundler](/guides/customizing-metro) has built-in support for optional imports. Optional imports refer to wrapping a `require` statement with a `try/catch` to prevent an error from being thrown when the requested module is missing:
 
-The above **MapView.js** would change to look like the following:
+<SnackInline dependencies={['expo-constants', 'react-native-blurhash']}>
 
 ```js
-import * as React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import React from 'react';
+import { View } from 'react-native';
 
-let MapView;
+let Blurhash;
 
 try {
-  const Mapbox = require('@react-native-mapbox-gl/maps').default;
-  Mapbox.setAccessToken('access-token-here');
-  MapView = Mapbox.MapView;
+  Blurhash = require('react-native-blurhash').Blurhash;
 } catch {
-  MapView = props => (
-    <View
-      style={[
-        {
-          backgroundColor: 'lightblue',
-          alignItems: 'center',
-          justifyContent: 'center',
-        },
-        props.style,
-      ]}>
-      <Text>🗺 (Mapbox not available)</Text>
-    </View>
-  );
+  Blurhash = View;
 }
 
-export default MapView;
+export default function App() {
+  return <Blurhash blurhash="LGFFaXYk^6#M@-5c,1J5@[or[Q6." style={{ flex: 1 }} />;
+}
 ```
 
-### **Deprecated**: use the `.expo.[js/json/ts/tsx]` extension to provide Expo Go specific fallbacks
+</SnackInline>
+
+This method is the least reliable because there are a number of reasons that a `require` statement might throw an error. There could be an internal error, the module could be missing, the native module could be linked incorrectly, etc. Ultimately you should try to avoid using this method.
+
+## **Deprecated**: use the `.expo.[js/json/ts/tsx]` extension to provide Expo Go specific fallbacks
 
 The `.expo` extension is removed in SDK 41. Consider using [conditional inline requires](#use-conditional-inline-requires-to-provide-fallbacks) instead, and read [expo.fyi/expo-extension-migration](https://expo.fyi/expo-extension-migration) for specific guidance on migrating away.
```

Comment:
```suggestion
The `.expo` extension is removed in SDK 41. Instead, consider using [conditional inline requires](#use-conditional-inline-requires-to-provide-fallbacks), and read [expo.fyi/expo-extension-migration](https://expo.fyi/expo-extension-migration) for specific guidance on migration.
```

Unrelated to PR, but I think this is a good opportunity to improve the verbiage here too.